fe_contracts/defi/src/raw_call.fe

```diff
@@ -21,40 +21,42 @@ pub struct RawCallHelper {
 
     /// Call a function directly by its selector on the given address. 
     pub fn call<T: MemoryEncoding>(_ addr: address, _ selector: u32, data: T) -> RawCallResult{
+        let mut start: u256
         unsafe {
             let input_offset: u256 = RawCallHelper::avail()
             [selector]
             data.write_mem()
-
+            start = RawCallHelper::avail() + data.byte_count()
             let success: u256 = std::evm::call(
                 gas: std::evm::gas_remaining(),
                 addr,
                 value: 0,
                 input_offset,
                 input_len: 4 + data.byte_count(),
-                output_offset: 0,
```

Comment:
Ah, sweet. I wasn't even aware the `call` could copy the return data directly. In fact, I took inspiration by how the Fe compiler handles external calls internally and it also uses `returndatacopy` manually: https://github.com/ethereum/fe/blob/2dcdaaa1a5663fe7eb14cfdb520144e1d5fdb959/crates/codegen/src/yul/runtime/contract.rs#L94-L104

How does this relate in terms of gas costs. Is leaving the copy to the `call` supposed to be more gas efficient compared to manually calling `returndatacopy`? What if the caller wasn't interested in the returned data at all, I guess the implicit copy would then unnecessarily increase the gas cost, no?

I'm just trying to understand the change better, I love the fact that you came here to improve my hacking :heart:

Reply:
I would imagine it's cheaper but I'd have to update the compiler codegen and experiment as it seems the compiler inserts stuff into unsafe blocks and I was treating it like inline Yul. Looking at it again, it does seem like the compiler inserts checks for returndatasize and the one I added is redundant. I was just making sure it'd revert if someone called an EOA. 

I think Vyper has a way to disable copying returndata and Solidity always copies it, but the latter results in some undesirable behavior if you're trying to prevent callee's from causing reverts https://github.com/ethereum/solidity/issues/12306